### PR TITLE
simplify some linq expressions

### PR DESCRIPTION
**Pull request by SimonCropp**
# Bug

Fixes: 

## PR Checklist

- [ ] Meaningful title, helpful description and a linked NuGet/Home issue
- [ ] Added tests
- [ ] Link to an issue or pull request to update docs if this PR changes settings, environment variables, new feature, etc.

### CODE REVIEW

File: src/NuGet.Core/NuGet.Commands/RestoreCommand/CompatibilityChecker.cs [hunk 1]
```diff
@@ -66,7 +66,7 @@ internal async Task<CompatibilityCheckResult> CheckAsync(
             if (packageSpec.RestoreMetadata?.ProjectStyle == ProjectStyle.DotnetToolReference)
             {
                 // Autoreferenced packages are allowed. Currently they're using Microsoft.NET.Platforms as an auto-ref package
-                if (packageSpec.GetAllPackageDependencies().Where(e => !e.AutoReferenced).Count() != 1)
+                if (packageSpec.GetAllPackageDependencies().Count(e => !e.AutoReferenced) != 1)
```
Comment (nkolev92) [inline src/NuGet.Core/NuGet.Commands/RestoreCommand/CompatibilityChecker.cs:69]: Seems like we can enable this rule to capture all these issues: 

https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0120

Can we enable: 
```
dotnet_diagnostic.IDE0120.severity = warning
```

likely on line 149 in https://github.com/NuGet/NuGet.Client/blob/dev/.editorconfig.

Comment (nkolev92) [reply src/NuGet.Core/NuGet.Commands/RestoreCommand/CompatibilityChecker.cs:69]: fwiw, enabling this does reveal some issues in NuGet.Frameworks\FrameworkReducer.cs for example.

Comment (SimonCropp) [reply src/NuGet.Core/NuGet.Commands/RestoreCommand/CompatibilityChecker.cs:69]: @nkolev92 done

**Comment by nkolev92** [comment]
Changes look good, there's some formatting issues the CI is reporting.

File: test/NuGet.Core.Tests/NuGet.Commands.Test/ProjectResolutionTests.cs [hunk 2]
```diff
@@ -261,13 +249,7 @@ public async Task ProjectResolution_ProjectReferenceWithInCompatibleTFM_Multiple
...
                 var logger = new TestLogger();
 
-                var restoreContext = new RestoreArgs()
-                {
-                    Sources = new List<string>() { packageSource.FullName },
-                    GlobalPackagesFolder = packagesDir.FullName,
-                    Log = logger,
-                    CacheContext = new SourceCacheContext()
-                };
+                var restoreContext = new RestoreArgs() {Sources = new List<string>() {packageSource.FullName}, GlobalPackagesFolder = packagesDir.FullName, Log = logger, CacheContext = new SourceCacheContext()};
```
Comment (zivkan) [inline test/NuGet.Core.Tests/NuGet.Commands.Test/ProjectResolutionTests.cs:252]: I don't like this change, I think it's much easier to read one property per line.  It's also unrelated to the title of this PR.

File: test/NuGet.Core.Tests/NuGet.Commands.Test/ProjectResolutionTests.cs [hunk 3]
```diff
@@ -165,13 +159,7 @@ public async Task ProjectResolution_ProjectReferenceWithInCompatibleTFM()
...
```
Comment (zivkan) [inline test/NuGet.Core.Tests/NuGet.Commands.Test/ProjectResolutionTests.cs:162]: I don't like this change, I think it's much easier to read one property per line.  It's also unrelated to the title of this PR.

File: test/NuGet.Core.Tests/NuGet.Commands.Test/ProjectResolutionTests.cs [hunk 4]
```diff
@@ -76,13 +76,7 @@ public async Task ProjectResolution_ProjectFrameworkAssemblyReferences()
...
```
Comment (zivkan) [inline test/NuGet.Core.Tests/NuGet.Commands.Test/ProjectResolutionTests.cs:79]: I don't like this change, I think it's much easier to read one property per line.  It's also unrelated to the title of this PR.

File: test/NuGet.Core.Tests/NuGet.Commands.Test/ProjectResolutionTests.cs [hunk 5]
```diff
@@ -351,13 +333,7 @@ public async Task ProjectResolution_ProjectReferenceWithInCompatibleTFMNoRange()
...
```
Comment (zivkan) [inline test/NuGet.Core.Tests/NuGet.Commands.Test/ProjectResolutionTests.cs:336]: I don't like this change, I think it's much easier to read one property per line.  It's also unrelated to the title of this PR.

File: test/NuGet.Core.Tests/NuGet.Commands.Test/ProjectResolutionTests.cs [hunk 6]
```diff
@@ -513,13 +489,7 @@ public async Task ProjectResolution_HigherVersionForProjectReferences()
...
```
Comment (zivkan) [inline test/NuGet.Core.Tests/NuGet.Commands.Test/ProjectResolutionTests.cs:492]: I don't like this change, I think it's much easier to read one property per line.  It's also unrelated to the title of this PR.

**Review by zivkan — COMMENTED** [review]
The LINQ changes are excellent.  The other changes are unnecessary 😁

**Comment by nkolev92** [comment]
@SimonCropp do you care if I reopen this PR and push some changes to your branch (assuming your fork allows it) and drive this PR to completion?

I think simplification changes are worth merging, and this way you'd get credit for the PR. 

Otherwise, I'd probably just recreate the PR from my own branch.

**Comment by SimonCropp** [comment]
@nkolev92 go for it. and i dont need credit. just make the changes in whatever way is easiest for you.

I only closed all my PRs since i found the NuGet.Client project to have too much unecessary friction to contribute to

**Comment by SimonCropp** [comment]
@nkolev92 if u want a look at a MS project that is easier to contribute to see testfx. 
https://github.com/microsoft/testfx/pulls?q=is%3Apr+author%3ASimonCropp+is%3Aclosed

**Pull request closed without merging**